### PR TITLE
Use destructuring more and minor tweaks

### PR DESCRIPTION
**Pull request by XhmikosR**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -65,33 +65,25 @@ function oust(src, type, raw = false) {
...
+  const chosenTypes = typeArray.map(type => ({...types[type], type}));
+  const $selector = $(chosenTypes.map(type => type.selector).join(', '));
 
-  return Array.prototype.map.call($(selector), element => {
-    const $element = $(element);
-    const chosenType = chosenTypes.find(type => $element.is(type.selector));
-
+  return [...$selector].map(element => {
+    const $el = $(element);
+    const {type, method, attribute} = chosenTypes.find(type => $el.is(type.selector));
```
Comment (bezoerb) [inline index.js:74]: this breaks if `chosenTypes.find` returns `undefined`
```suggestion
    const {type, method, attribute} = chosenTypes.find(type => $el.is(type.selector)) || {};
```

Comment (XhmikosR) [reply index.js:74]: Thanks! I had the same feeling too, but I wasn't sure if it does indeed return undefined since we don't seem to have a test for it.

**Comment by XhmikosR** [comment]
@bezoerb if you could also add a test so that we cover the undefined case that would be perfect :)

**Comment by bezoerb** [comment]
@XhmikosR: just took a closer look and i think we don't need the `|| {}` because as of my understanding it can't return `undefined` as we're only looping over the elements which have been previously selected.

I thought one might just pass an invalid type to the function but this is also already handled.

**Review by bezoerb — APPROVED** [review]

**Comment by XhmikosR** [comment]
I just reverted the fallback addition, can you confirm all is good?

**Review by bezoerb — APPROVED** [review]

**Pull request merged**